sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/TextAnalyticsClientLiveTestBase.cs

```diff
@@ -43,5 +51,23 @@ public TextAnalyticsClientLiveTestBase(bool isAsync, TextAnalyticsClientOptions.
                 return InstrumentClient(new TextAnalyticsClient(endpoint, credential, InstrumentClientOptions(options)));
             }
         }
+
+        // This has been added to stop the custom tests to run forever while we
+        // get more reliable information on which scenarios cause timeouts.
+        // Issue https://github.com/Azure/azure-sdk-for-net/issues/25152
+        internal async Task PollUntilTime(AnalyzeActionsOperation operation)
+        {
+            TimeSpan pollingInterval = TimeSpan.FromSeconds(10);
+            int timeout = (int)TimeSpan.FromMinutes(20).TotalMilliseconds;
+            using CancellationTokenSource cts = new CancellationTokenSource(timeout);
```

Comment:
nit: `CancellationTokenSource` has a constructor that takes a `TimeSpan` directly, so there's no need to convert to milliseconds. On the other hand, we'll remove this piece of code eventually, so if you don't have any other changes to do to this PR, I think it's not worth updating.

Reply:
updated